### PR TITLE
add file for milter headers

### PR DESCRIPTION
**Pull request by fabianfrz**
also contains a bugfix

### CODE REVIEW

File: mail/rspamd/src/opnsense/mvc/app/controllers/OPNsense/Rspamd/forms/settings.xml [hunk 1]
```diff
@@ -20,9 +20,54 @@
...
+                <id>rspamd.milter_headers.enabled</id>
+                <label>Enable</label>
+                <type>checkbox</type>
+                <help>When this checkbox is enabled, you can add additional headers to your mails, which contain information about the scan results.</help>
+            </field>
+            <field>
+                <id>rspamd.milter_headers.enable_authentication_results</id>
+                <label>Enable Authentication Result Headers</label>
+                <type>checkbox</type>
+                <help>If this checkbox is enabled, a header about the authentication state</help>
```
Comment (mimugmail) [inline mail/rspamd/src/opnsense/mvc/app/controllers/OPNsense/Rspamd/forms/settings.xml]: Something Missing here

File: mail/rspamd/src/opnsense/mvc/app/controllers/OPNsense/Rspamd/forms/settings.xml [hunk 2]
```diff
@@ -20,9 +20,54 @@
...
+            </field>
+            <field>
+                <id>rspamd.milter_headers.enable_spamd_bar</id>
+                <label>Enable Spamd Bar</label>
+                <type>checkbox</type>
+                <help>Add a header containing a visible indicator of the spam level.</help>
+            </field>
+            <field>
+                <id>rspamd.milter_headers.skip_local</id>
+                <label>Skip Local</label>
```
Comment (mimugmail) [inline mail/rspamd/src/opnsense/mvc/app/controllers/OPNsense/Rspamd/forms/settings.xml]: What means skip local

File: mail/rspamd/src/opnsense/mvc/app/models/OPNsense/Rspamd/RSpamd.xml [hunk 3]
```diff
@@ -13,6 +13,37 @@
...
+      <skip_local type="BooleanField">
+        <default>0</default>
+        <Required>Y</Required>
+      </skip_local>
+      <skip_authenticated type="BooleanField">
+        <default>0</default>
+        <Required>Y</Required>
+      </skip_authenticated>
+      <extended_headers_rcpt type="CSVListField">
+        <mask>/[a-z0-9\.\-@,]+/i</mask>
```
Comment (mimugmail) [inline mail/rspamd/src/opnsense/mvc/app/models/OPNsense/Rspamd/RSpamd.xml]: Perhaps also _ for something in fron of @?

File: mail/rspamd/src/opnsense/mvc/app/controllers/OPNsense/Rspamd/forms/settings.xml (hunk 2 above)
Comment (fabianfrz) [reply mail/rspamd/src/opnsense/mvc/app/controllers/OPNsense/Rspamd/forms/settings.xml]: [Docs](https://rspamd.com/doc/modules/milter_headers.html) say:
"Set false to always add headers for local IPs (default true)"

File: mail/rspamd/src/opnsense/mvc/app/models/OPNsense/Rspamd/RSpamd.xml (hunk 3 above)
Comment (fabianfrz) [reply mail/rspamd/src/opnsense/mvc/app/models/OPNsense/Rspamd/RSpamd.xml]: is that legit? Have never seen that.

Comment (mimugmail) [reply mail/rspamd/src/opnsense/mvc/app/models/OPNsense/Rspamd/RSpamd.xml]: Yep, we also use this company wide for distribution lists

**Comment by fabianfrz** [comment]
@mimugmail you feedback is added

**Pull request merged**